nixos/modules/programs/zsh/zsh.nix

```diff
@@ -187,6 +187,9 @@ in
             fpath=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions $p/share/zsh/vendor-completions $fpath)
         done
 
+        # Alternative method of determining short and full hostname.
```

Comment:
I can confirm this issue exists on NixOS, but I do not think the zsh module is the right place to fix it.  From the source code of zsh, it sets `HOST` to the `nodename` of `uname`.

Do other distributions have this issue?  If not, how do they fix it or maybe our way of setting fqdn can be improved?

Reply:
I asked on their IRC channel, they recommended using a variable and putting the result of the `hostname -f` command into it.
But the `hostname -f` command does not work for us. For correct working it is necessary to disable the `nscd` service.
The `$HOST` variable is not used elsewhere, so I just overwrote it.

> Do other distributions have this issue? If not, how do they fix it or maybe our way of setting fqdn can be improved?

Checked on Ubuntu, doesn't work either.
Previously, the FQDN name was taken from the /etc/hostname file. Now it doesn't work.